### PR TITLE
feat(GithuService (76 > main)) GitHub OAuth 2.0

### LINKED ISSUE #76: Service Github
Connect to github api, search how to connect with OAuth 2.0

### PR DESCRIPTION
**Pull request by Pablodeibar**
Closes #76 
- create a service table with Github as enum and its access token
- update README + docker-compose

### CODE REVIEW

**Review by AlexGuillard — COMMENTED** [review]
seems ok

File: server/src/github-auth/github-auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  @Get()
+  @UseGuards(AuthGuard('github'))
+  async login() {
+    // initiates the GitHub OAuth2 login flow
+  }
+
+  @Get('callback')
+  @UseGuards(AuthGuard('github'))
+  async authCallback(@Req() req, @Res() res) {
```
Comment (Benjicatch) [inline server/src/github-auth/github-auth.controller.ts:21]: put this logic in a service file + remove console.log

**Review by Benjicatch — CHANGES_REQUESTED** [review]

**Review by Benjicatch — APPROVED** [review]
ok

**Review by Benjicatch — DISMISSED** [review]
ok

**Review by Benjicatch — DISMISSED** [review]
ok

**Review by Benjicatch — APPROVED** [review]
ok

File: server/src/github-auth/github-auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const user = req.user;
+
+    try {
+      await this.githubAuthService.handleGithubAuthCallback(user);
+      return res.redirect('http://localhost:8081/Area');
```
Comment (Miou-zora) [inline server/src/github-auth/github-auth.controller.ts:24]: Can you use env var: if we have the time (I wish) to deploy this, we will maybe not gonna use this same ip / port.

File: server/src/github-auth/github-auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    } catch (error) {
+      throw error;
```
Comment (Miou-zora) [inline server/src/github-auth/github-auth.controller.ts:26]: You need to catch and throw again ?

File: server/src/strategy/github.strategy.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { PassportStrategy } from '@nestjs/passport';
+import { Profile, Strategy } from 'passport-github';
+
+@Injectable()
+export class GithubStrategy extends PassportStrategy(Strategy, 'github') {
+  constructor(configService: ConfigService) {
+    super({
+      clientID: configService.get<string>('GITHUB_CLIENT_ID'),
+      clientSecret: configService.get<string>('GITHUB_CLIENT_SECRET'),
+      callbackURL: 'http://localhost:8080/auth/github/callback',
```
Comment (Miou-zora) [inline server/src/strategy/github.strategy.ts:12]: Same as previous review mark: use env var for ip / port

**Review by Miou-zora — CHANGES_REQUESTED** [review]

File: server/src/strategy/github.strategy.ts (hunk 4 above)
Comment (Pablodeibar) [reply server/src/strategy/github.strategy.ts]: fixed but we also need to change on github OAuth app if we deploy

File: server/src/github-auth/github-auth.controller.ts (hunk 2 above)
Comment (Pablodeibar) [reply server/src/github-auth/github-auth.controller.ts]: fixed but we also need to change on github OAuth app if we deploy

File: server/src/github-auth/github-auth.controller.ts (hunk 3 above)
Comment (Pablodeibar) [reply server/src/github-auth/github-auth.controller.ts:26]: i can do a console log if you want but if i do a try i do a catch too, i dont know what else to do

**Review by AlexGuillard — DISMISSED** [review]
ok

**Review by Benjicatch — APPROVED** [review]
ok

**Review by Benjicatch — APPROVED** [review]
ok

**Review by Miou-zora — APPROVED** [review]

File: docker-compose.dev.yml [hunk 5]
```diff
@@ -30,6 +30,8 @@ services:
       WEB_URL: ${WEB_URL}
       REFRESH_TOKEN: ${REFRESH_TOKEN}
       EMAIL: ${EMAIL}
+      GITHUB_CLIENT_ID: ${GITHUB_CLIENT_ID}
+      GITHUB_CLIENT_SECRET: ${GITHUB_CLIENT_SECRET}
```
Comment (Miou-zora) [inline docker-compose.dev.yml:34]: Didi you forget `BASE_URL: ${BASE_URL}`  here?

File: README.md [hunk 6]
```diff
@@ -14,6 +14,8 @@ SERVER_URL=Url of the server like 'http://localhost:8080'
 WEB_URL=Url of the web app like 'http://localhost:8081'
 GOOGLE_CLIENT_ID=Your GOOGLE_CLIENT_ID
 GOOGLE_CLIENT_SECRET=Your GOOGLE_CLIENT_SECRET
+GITHUB_CLIENT_ID=Your GITHUB_CLIENT_ID
+GITHUB_CLIENT_SECRET=Your GITHUB_CLIENT_SECRET
```
Comment (Miou-zora) [inline README.md:18]: Did you forget to add `BASE_URL` ?

File: docker-compose.dev.yml (hunk 5 above)
Comment (Pablodeibar) [reply docker-compose.dev.yml:34]: fixed

File: README.md (hunk 6 above)
Comment (Pablodeibar) [reply README.md:18]: fixed

**Review by Miou-zora — APPROVED** [review]

**Review by Benjicatch — APPROVED** [review]
ok

**Review by AlexGuillard — APPROVED** [review]
ok

**Pull request merged**